src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEAFULT_SIZE = 8;
+    private Object[] stack;
+    private int size = 0;
 
+    public MateStack() {
+        stack = new Object[DEAFULT_SIZE];
+        Arrays.fill(stack, null);
+    }
+
+    public void push(T value) throws RuntimeException {
```

Comment:
The 'push' method should not throw a RuntimeException. Instead, it should handle the situation where the stack is full by resizing the stack array to accommodate more elements. This is a misunderstanding of the task requirements, as typically, a stack should dynamically resize when full.